cache/httpproxy/caching_http_proxy.go

```diff
@@ -37,15 +37,16 @@ type CacheableResponse struct {
 
 func (p cachingHttpProxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	// Rewrite the request URL to new host
-	targetURL, _ := url.JoinPath(p.targetHost, req.URL.RequestURI())
-	req, err := http.NewRequestWithContext(req.Context(), req.Method, targetURL, req.Body)
+	targetURL, _ := url.JoinPath(p.targetHost, req.URL.Path)
+	proxyReq, err := http.NewRequestWithContext(req.Context(), req.Method, targetURL, req.Body)
 	if err != nil {
 		panic(fmt.Sprintf("error rewriting request to %s %s: %v", req.Method, req.URL.String(), err))
 	}
+	proxyReq.URL.RawQuery = req.URL.Query().Encode()
```

Comment:
what is this, we update the url after making the request?

Reply:
It just adds the query params properly to the new proxied request. The proxy request is then queried below in `GetFromCacheOrCall`